src/main/java/lotto/model/ScoreMap.java

```diff
@@ -0,0 +1,62 @@
+package lotto.model;
+
+import lotto.model.config.LottoConfig;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class ScoreMap {
+    private Map<Integer, Integer> scoreMap;
+
+    public ScoreMap() {
+        this.scoreMap = new HashMap<Integer, Integer>(){{
+            put(0, 0);
+            put(1, 0);
+            put(2, 0);
+            put(3, 0);
+            put(4, 0);
+            put(5, 0);
+            put(6, 0);
+        }};
+    }
+
+    public Set<Map.Entry<Integer, Integer>> getEntrySet(){
+        return scoreMap.entrySet();
+    }
+
+    public void updateWithNumMatched(int numMatched) {
+        scoreMap.put(numMatched, scoreMap.get(numMatched) + 1);
+    }
+
+    public int calculateReward() {
+        int totalProfit = 0;
+        for( Map.Entry<Integer, Integer> lottoResult : scoreMap.entrySet() ){
```

Comment:
```suggestion
        for (Map.Entry<Integer, Integer> lottoResult : scoreMap.entrySet()) {
```
`LottoConfig`에서도 의견 드렸지만 `Map`이기 때문에 계속적으로 연산비용이 발생합니다 😭 
개선 검토 부탁 드리겠습니다 🙇

Reply:
이 부분은 잘 이해하지 못하였는데,  
keySet() 메소드로 순회하도록 고쳐보았습니다.